staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer.go

```diff
@@ -134,6 +137,28 @@ func (g *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, svc
 		return nil, err
 	}
 
+	// Services with multiples protocols are not supported by this controller, warn the users and sets
+	// the corresponding Service Status Condition.
+	// https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1435-mixed-protocol-lb
+	if err := checkMixedProtocol(svc.Spec.Ports); err != nil {
+		if hasLoadBalancerPortsError(svc) {
+			return nil, err
```

Comment:
is returning an error here correct? that bubbles up to requeue the service for reprocessing, which will just fail again

```
c.serviceQueue.AddRateLimited(key)
```

Reply:
I have the same doubt and actually, my first approach was to return nil, but since we don't know what the caller is going to do after this I considered safer to return an error since this operations is not going to succeed.

Controller receives a new services and calls EnsureLoadBalancers, it returns an error and patch the Services, the errors is requeued, and the same Service calls EnsureLoadBalancer but now it returns nil.

If we can guarantee that the caller will check the condition I think returning nil will be safe, but otherwise, I can't think other way of indicating that EnsureLoadBalancer didn't ensured nothing and the loadbalancer does not exist

Reply:
Agree with the question.  What is the behavior we want?

Retrying will fail and fail and fail - and not even repeat the event IIUC.    I don't see any point in that, vs. just saying we succeeded and setting the error.

Alternately, return the error and re-assert the event and re-apply the condition (just to be sure it doesn't get removed), the same as an error from the clouid API?

Reply:
> Retrying will fail and fail and fail - and not even repeat the event IIUC. I don't see any point in that, vs. just saying we succeeded and setting the error.

the retry is bounded , the controller does not retry forever

Reply:
I agree - my real question is whether we should re-assert the event (since those expire) every failed attempt.

Reply:
I'm not fan of event flooding as a mechanism to warn users, this just send one event and update the service status , I think that is enough for any monitoring system to work with.

Reply:
Given that we WANT to implement this eventually (soon?) I am OK with that.

Reply:
oh, hehe,  that is an important details. we are not going to touch this code anymore, this is for backports, the new implementation has to go here https://github.com/kubernetes/cloud-provider-gcp/blob/master/providers/gce/gce_loadbalancer.go

gke is going to switch to that in 1.27 IIRC